[Li Siqi] iP

siqirua opened the pull request:
# RuaPro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Note that this md is used merely for the intention of practicing GFMD and it does not represent the actual product.

siqirua committed:
Change Chatbot name and beginning message

siqirua committed:
Add Echo Feature

siqirua committed:
Add list feature and create TaskList and Task class

siqirua committed:
Add mark feature

siqirua committed:
Add Task Type and implement 3 subclass of Task

siqirua committed:
Edit the SUT

siqirua committed:
Add two exceptions

siqirua committed:
Add delete feature

siqirua committed:
Edit the expected.txt

siqirua committed:
Merge commit '53a652449724fb82b59d5384dc000f41897686cf'

siqirua committed:
Add save feature

siqirua committed:
Merge branch 'branch-Level-7'

siqirua committed:
Change the time from String type to LocalDate type

siqirua committed:
Merge branch 'branch-Level-8'

siqirua committed:
Add feature of searching by date

siqirua committed:
Merge branch 'branch-Level-8'

siqirua committed:
Unfinished edit to include more OOP

siqirua committed:
Include more OOP

siqirua committed:
Merge branch 'branch-more-oop'

siqirua committed:
Organize the classes into suitable java packages.

siqirua committed:
Set up gradle

siqirua committed:
Include 2 JUnit Test

siqirua committed:
Add JavaDoc for some methods

siqirua committed:
Add JavaDoc to some methods

siqirua committed:
Tweak the code to comply with a coding standard

siqirua committed:
Implement keyword search function

siqirua committed:
Merge branch 'branch-A-JavaDoc'

siqirua committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/rua/Rua.java
#	src/main/java/rua/command/ClearCommand.java
#	src/main/java/rua/command/Command.java
#	src/main/java/rua/common/Storage.java
#	src/main/java/rua/exception/EmptyDescriptionException.java
#	src/main/java/rua/exception/InvalidCommandException.java
#	src/main/java/rua/exception/InvalidTypeException.java
#	src/main/java/rua/task/Event.java
#	src/main/java/rua/task/Task.java
#	src/main/java/rua/task/TaskList.java
#	src/main/java/rua/task/Todo.java

siqirua committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/rua/common/Parser.java
#	src/main/java/rua/task/TaskList.java

siqirua committed:
Use checkStyle to detect coding style violations

siqirua committed:
Create a GUI for this project

siqirua committed:
Improve GUI further

siqirua committed:
Implement GUI partially

siqirua committed:
Update README.md

siqirua committed:
Merge pull request #1 from siqirua/patch-1

Update README.md

KumChaiYin commented on src/main/java/rua/command/MarkCommand.java:
```diff
@@ -0,0 +1,65 @@
+import rua.task.TaskList;
+
+public class MarkCommand implements Command {
+    private final Boolean marked;
```
Should this be `isMarked`?
```suggestion
    private final Boolean isMarked;
```

KumChaiYin commented on src/main/java/rua/task/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+     * @param task A Task object that will be added into the TaskList.
+     * @return The new TaskList object with the new Task.
+     */
+    public TaskList add(Task task) {
```
Perhaps can make the method name more specific
```suggestion
    public TaskList addTask(Task task) {
```

limzhenwy commented on src/main/data/Rua_copy.java:
```diff
@@ -0,0 +1,119 @@
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+
+        String greeting = "____________________________________________________________\n" +
```
Maybe you could consider adding access modifiers for 'greeting' and 'goodbye'. 

You could also add 'final' since these Strings will not be changed.
```suggestion
        private final String greeting = "____________________________________________________________\n" +
```

KumChaiYin commented on src/main/java/rua/common/Parser.java:
```diff
@@ -0,0 +1,116 @@
+     * @throws InvalidCommandException if the message is not supported.
+     */
+    public static Command parse(String message) throws EmptyDescriptionException, InvalidCommandException {
+        String[] inputs = message.split(" ", 2);
```
I like how you attached array specifiers to the type (not the variable).

limzhenwy commented on src/main/java/rua/command/Command.java:
```diff
@@ -0,0 +1,33 @@
+import rua.common.Ui;
+import rua.task.TaskList;
+
+public interface Command {
```
ingenious interface 👍

limzhenwy commented on src/main/java/rua/task/Todo.java:
```diff
@@ -0,0 +1,86 @@
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
in general, across the different task types, you could consider just altering the isMarked boolean, instead of needing to create new objects whenever isMarked is changed.

KumChaiYin commented on src/main/java/rua/command/AddCommand.java:
```diff
@@ -0,0 +1,64 @@
+     * @return A boolean indicating whether they are equal or not.
+     */
+    @Override
+    public boolean equals(Object o) {
```
I like how you separate the logical units within a block by one blank line.

KumChaiYin commented on src/main/java/rua/Rua.java:
```diff
@@ -0,0 +1,58 @@
+import rua.common.Ui;
+import rua.task.TaskList;
+
+public class Rua {
```
Would you like to add descriptive header comments for all public classes?

KumChaiYin commented on src/main/java/rua/Rua.java:
```diff
@@ -0,0 +1,58 @@
+                Command c = Parser.parse(fullCommand);
+                tasks = c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } catch (Exception e) {
```
Should it has few lines of catch statements, instead of simply catching Exception (Pokemon Exception Handling)?

KumChaiYin reviewed (commented):
Looks good. 👍 
There isn't any major coding standard related issues and I like how you used the white space properly:
All the Java reserved words are followed by a white space.
Operators are surrounded by a space character.
I have only a few suggestions.

siqirua committed:
Update README.md

siqirua committed:
Implement GUI further

siqirua committed:
Merge branch 'branch-Level-10'

siqirua committed:
Include some assertions in Event and TaskList files

Currently, the "mark", "unmark" and "delete" command is unchecked and
may cause potential errors. To address this problem, let's include some
assertions to make sure that the index given is no more than the size of
current task list.

siqirua committed:
Improve code quality

siqirua committed:
Improve code quality further

Currently the parser class only has one chunky method and there are
quite a number of magic strings, to improve the readability, let's apply
SLAP and refractor codes.

siqirua committed:
Merge pull request #3 from siqirua/branch-A-CodeQuality

Merge with branch a code quality

siqirua committed:
Merge pull request #2 from siqirua/branch-A-Assertions

Include some assertions in Event and TaskList files

siqirua committed:
Implement tagging feature partially

siqirua committed:
Implement Tagging feature

Currently we do not have tagging system and it would be difficult to see
the grouping of the tasks. Let's implement adding, deleting and clearing
tag functions in our chatbot!

siqirua committed:
Merge branch 'branch-C-Tagging'

siqirua committed:
Improve code quality

siqirua committed:
Add a representative screenshot of the product

siqirua committed:
Add a User Guide

siqirua committed:
Fix some bugs related to saving

siqirua committed:
Use date and time to denote the due time and start and end time of
deadline and event

siqirua committed:
Fix some typos in README.md
